src/sentry/ingest/outcomes_consumer.py

```diff
@@ -59,56 +62,68 @@ def is_signal_sent(project_id, event_id):
     return cache.get(key, None) is not None
 
 
-class OutcomesConsumerWorker(AbstractBatchWorker):
-    def process_message(self, message):
-        msg = json.loads(message.value())
+def _process_message(message):
+    msg = json.loads(message)
+
+    project_id = int(msg.get("project_id", 0))
+    if project_id == 0:
+        return  # no project. this is valid, so ignore silently.
+
+    outcome = int(msg.get("outcome", -1))
+    if outcome not in (Outcome.FILTERED, Outcome.RATE_LIMITED):
+        return  # nothing to do here
 
-        project_id = int(msg.get("project_id", 0))
-        if project_id == 0:
-            return True  # no project. this is valid, so ignore silently.
+    event_id = msg.get("event_id")
+    if is_signal_sent(project_id=project_id, event_id=event_id):
+        return  # message already processed nothing left to do
 
-        outcome = int(msg.get("outcome", -1))
-        if outcome not in (Outcome.FILTERED, Outcome.RATE_LIMITED):
-            return True  # nothing to do here
+    try:
+        project = Project.objects.get_from_cache(id=project_id)
+    except Project.DoesNotExist:
+        logger.error("OutcomesConsumer could not find project with id: %s", project_id)
+        return
 
-        event_id = msg.get("event_id")
-        if is_signal_sent(project_id=project_id, event_id=event_id):
-            return True  # message already processed nothing left to do
+    reason = msg.get("reason")
+    remote_addr = msg.get("remote_addr")
 
-        try:
-            project = Project.objects.get_from_cache(id=project_id)
-        except Project.DoesNotExist:
-            logger.error("OutcomesConsumer could not find project with id: %s", project_id)
-            return True
+    if outcome == Outcome.FILTERED:
+        event_filtered.send_robust(ip=remote_addr, project=project, sender=OutcomesConsumerWorker)
+    elif outcome == Outcome.RATE_LIMITED:
+        event_dropped.send_robust(
+            ip=remote_addr, project=project, reason_code=reason, sender=OutcomesConsumerWorker
+        )
 
-        reason = msg.get("reason")
-        remote_addr = msg.get("remote_addr")
+    # remember that we sent the signal just in case the processor dies before
+    mark_signal_sent(project_id=project_id, event_id=event_id)
 
-        if outcome == Outcome.FILTERED:
-            event_filtered.send_robust(ip=remote_addr, project=project, sender=self.process_message)
-        elif outcome == Outcome.RATE_LIMITED:
-            event_dropped.send_robust(
-                ip=remote_addr, project=project, reason_code=reason, sender=self.process_message
-            )
 
-        # remember that we sent the signal just in case the processor dies before
-        mark_signal_sent(project_id=project_id, event_id=event_id)
+class OutcomesConsumerWorker(AbstractBatchWorker):
+    def __init__(self, multiprocessing, concurrency):
```

Comment:
Do we not use `concurrency` anywhere now?

Reply:
`concurrency` is not an option for batchingkafkaconsumer if that's what you're asking.

Reply:
Ah, you're right. Fixed. (Also tested properly this time)

Reply:
👍